support-frontend/assets/helpers/redux/checkout/product/selectors/isSupporterPlus.ts

```diff
@@ -1,10 +1,37 @@
+import type {
+	ContributionType,
+	OtherAmounts,
+	SelectedAmounts,
+} from 'helpers/contributions';
 import { getAmount } from 'helpers/contributions';
+import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
 import { getContributionType } from 'helpers/redux/checkout/product/selectors/productType';
 import type { ContributionsState } from 'helpers/redux/contributionsStore';
 import { getThresholdPrice } from 'helpers/supporterPlus/benefitsThreshold';
 import { isOneOff } from 'helpers/supporterPlus/isContributionRecurring';
 
-export function isSupporterPlusPurchase(state: ContributionsState): boolean {
+export function isSupporterPlus(
+	contributionType: ContributionType,
+	selectedAmounts: SelectedAmounts,
+	otherAmounts: OtherAmounts,
+	countryGroupId: CountryGroupId,
+): boolean {
+	if (isOneOff(contributionType)) {
+		return false;
+	}
+
+	const benefitsThreshold = getThresholdPrice(countryGroupId, contributionType);
+	const selectedAmount = selectedAmounts[contributionType];
+
+	if (selectedAmount === 'other') {
+		const otherAmount = otherAmounts[contributionType].amount;
+		return otherAmount ? parseInt(otherAmount) >= benefitsThreshold : false;
+	}
```

Comment:
What is the difference between this and the `isSupporterPlusFromState` calculation?